packages/react/src/components/ListItem/ListItem.tsx

```diff
@@ -11,7 +11,7 @@ import classnames from 'classnames';
 import { usePrefix } from '../../internal/usePrefix';
 import { Text } from '../Text';
 
-type ListItemProps = ComponentProps<'li'>;
+type ListItemProps = Omit<ComponentProps<'li'>, 'ref'>;
```

Comment:
Added this fix because of this typescript error:
`Types of property 'ref' are incompatible.`
This removes the ref property from `ComponentProps<'li'>`, preventing conflicts with the Text component.